flink-1.17-connector-bigquery/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/table/BigQueryDynamicTableSinkTest.java

```diff
@@ -0,0 +1,157 @@
+    }
+
+    @Test
+    public void testConstructor() {
+        BigQuerySinkConfig obtainedSinkConfig = bigQueryDynamicTableSink.getSinkConfig();
+        assertEquals(logicalTypeSchema, bigQueryDynamicTableSink.getLogicalType());
+        assertEquals(DeliveryGuarantee.AT_LEAST_ONCE, obtainedSinkConfig.getDeliveryGuarantee());
+        Schema convertedAvroSchema =
+                new Schema.Parser()
+                        .parse(
+                                "{\"type\":\"record\",\"name\":\"record\","
+                                        + "\"namespace\":\"org.apache.flink.avro.generated\",\"fields\":"
+                                        + "[{\"name\":\"number\",\"type\":\"long\"}]}");
+        assertEquals(convertedAvroSchema, obtainedSinkConfig.getSchemaProvider().getAvroSchema());
+    }
+
+    @Test
+    public void testCopy() {
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        BigQueryDynamicTableSink bigQueryDynamicTableSinkCopy =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+
+        assertTrue(checkEquality(bigQueryDynamicTableSinkCopy, bigQueryDynamicTableSink.copy()));
+    }
+
+    @Test
+    public void testSummaryString() {
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        assertEquals("BigQuery", bigQueryDynamicTableSink.asSummaryString());
+    }
+
+    @Test
+    public void testChangelogMode() {
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        assertEquals(
+                ChangelogMode.insertOnly(),
+                bigQueryDynamicTableSink.getChangelogMode(Mockito.mock(ChangelogMode.class)));
+    }
+
+    @Test
+    public void testSinkRuntimeProvider() {
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        assertInstanceOf(
+                SinkV2Provider.class,
+                bigQueryDynamicTableSink.getSinkRuntimeProvider(
+                        Mockito.mock(DynamicTableSink.Context.class)));
+    }
+
+    /**
+     * Method overwritten to check equality, required for testing.
+     *
+     * @param bigQueryDynamicSink1 Object 1.
+     * @param bigQueryDynamicSink2 Target Object to check equality with.
+     * @return True if {@link Object} is equal to current object.
+     */
+    private boolean checkEquality(Object bigQueryDynamicSink1, Object bigQueryDynamicSink2) {
+        if (bigQueryDynamicSink1 == bigQueryDynamicSink2) {
+            return true;
+        }
+        if (bigQueryDynamicSink1.getClass() != bigQueryDynamicSink2.getClass()) {
+            return false;
+        }
+        BigQueryDynamicTableSink bigQueryDynamicTableSink1 =
```

Comment:
Looking at this implementation it looks like the equality is mainly on the config and not the actual sink object. Why not override the equals method on the config class itself?

Reply:
ACK, moved to the class.